controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -314,6 +325,30 @@ public OperationContext createStreamContext(String scopeName, String streamName,
         }), executor);
     }
 
+    /**
```

Comment:
@shshashwat This method is just a duplicate of the existing `deleteScope` method. Please remove this and use the existing deleteScope method instead.

Reply:
From `deleteScope()` we are calling Scope's `deleteScope()` method while from `deleteScopeRecursive()` we are calling Scope's `deleteScopeRecursive()`. `deleteScopeRecursive()` does an additional task of deleting scope from `deleting_scopes_table`

Reply:
Rather than duplicate 90% of the code in these 2 methods, I'd prefer doing the following in DeleteScopeTask:

`streamMetadataStore.deleteScope(...).thenCompose(v -> streamMetadataStore.removeFromDeletingScopesTable)`

Reply:
Refactoring can be done to extract the common part here, but I would prefer to call `removeEntry` from outside `DeleteScopeTask`. Also, with this approach we will need to create a new method if that's your suggestion.